Feat/extend get language icon

zerob13 opened the pull request:
- support extend getLanguageIcon #28 
- support expend code block on loading
- show the expend button for all code block for better perf

zerob13 committed:
feat: add support for custom languageicon #28

zerob13 committed:
fix: expand code on loading is false

Simon-He95 commented:
<img width="263" height="108" alt="image" src="https://github.com/user-attachments/assets/814d85fb-9720-42fd-aedb-9aa17131e14a" />

The full screen button is broken. First, it shouldn't appear when there's no scroll bar. Second, it should appear when the font size is enlarged and the scroll bar is visible. The full screen button should also disappear when the font size is reduced.

zerob13 committed:
chore: Merge branch 'main' into feat/extend-getLanguageIcon

zerob13 committed:
chore: resolve merge conflicts from main branch

Merged changes from main branch while preserving both feature branch
and main branch logic. Resolved conflicts in:
- playground/src/main.ts: kept VueRendererMarkdown import
- src/components/CodeBlockNode/CodeBlockNode.vue: merged both async loading
  logic and Mermaid cleanup logic
